format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingInputFormatProvider.java

```diff
@@ -63,6 +65,26 @@ protected void validate() {
     // no-op
   }
 
+  protected Schema addPathField(Schema dataSchema, FailureCollector failureCollector) {
+    List<Schema.Field> fields = new ArrayList<>(dataSchema.getFields());
+    addPathField(fields, failureCollector);
+    return Schema.recordOf(dataSchema.getRecordName(), fields);
+  }
+
+  protected void addPathField(List<Schema.Field> dataFields, FailureCollector failureCollector) {
+    String pathField = conf.getPathField();
+    if (pathField != null) {
+      for (Schema.Field field : dataFields) {
+        if (pathField.equals(field.getName())) {
+          failureCollector.addFailure("Pathfield value already exists in the schema",
+            String.format("Please provide a new value instead of %s", pathField));
```

Comment:
should be adding a cause with the path field as the problem property. This will result in the UI highlighting that field and displaying the error message under it.

```
failureCollector.addFailure(...)
  .withConfigProperty("pathField")
```

Reply:
Added